src/Corax/Queries/SortingMatch/SortingMatch.cs

```diff
@@ -247,9 +281,169 @@ void AddDocumentBoost(ref SortingMatch<TInner, TComparer> match, Span<long> matc
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int ExecuteSorting<TOut>(ref SortingMatch<TInner, TComparer> match, Span<long> matches)
+            where TOut : struct
+        {
+            // PERF: This method assumes that we need to perform sorting of the whole set (even if we have a take statement).
+            // However, there is space to improve this further doing iterative sorting with element discarding in cases where
+            // take statements requires us to remove lots of elements. While we are not going to perform this optimization yet
+            // sorting is a costly primitive that if we can avoid it, would provide us a great opportunity for improvement. 
+
+            int totalMatches = (int)match.TotalResults;
+
+            int itemArraySize = Unsafe.SizeOf<MatchComparer<TComparer, TOut>.Item>() * totalMatches;
+            using var _ = match._searcher.Allocator.Allocate(itemArraySize, out var bufferHolder);
+
+            var itemKeys = MemoryMarshal.Cast<byte, MatchComparer<TComparer, TOut>.Item>(bufferHolder.ToSpan().Slice(0, itemArraySize));
+            Debug.Assert(itemKeys.Length == totalMatches);
+
+            var searcher = match._searcher;
+            var field = match._comparer.Field;
+            var comparer = new MatchComparer<TComparer, TOut>(match._comparer);
+            for (int i = 0; i < totalMatches; i++)
+            {
+                var read = Get(searcher, field, matches[i], out itemKeys[i].Value, match._comparer);
+                itemKeys[i].Key = read ? matches[i] : -matches[i];
+            }
+
+            // We sort the the set
+            var sorter = new Sorter<MatchComparer<TComparer, TOut>.Item, long, MatchComparer<TComparer, TOut>>(comparer);
+            sorter.Sort(itemKeys, matches[0..totalMatches]);
+
+            // We have a take statement so we are only going to care about the highest priority elements. 
+            if (match._take > 0)
+                totalMatches = Math.Min(totalMatches, match._take);
```

Comment:
@maciejaszyk this will get executed and restrict the amount of actual values to return. The reason is that query builder is assuming that no filtering at Corax level happens but it is setting the take value to perform it. This will cause tests like `CanUseFilterQueryOnMapIndexes` to fail when in reality the query is not setting the `take` parameter.

Query in question:

```
var emp = s.Advanced.RawQuery<Employee>("from Employees order by Name filter Active = false")
    .Statistics(out var stats)
    .SingleOrDefault();
```

Can you take a look why this is happening?